tests/lib/state.sh

```diff
@@ -80,6 +88,13 @@ restore_snapd_state() {
 
         tar -C/ -xf "$SNAPD_STATE_FILE"
     fi
+
+    # Start all the units which have to be active
+    for unit in $(cat "$SNAPD_ACTIVE_UNITS"); do
```

Comment:
I think you may need to filter out some services here. For sure `snapd.snap-repair.service`, because it's started by the timer.